test/suite/test_prepare_discover01.py

```diff
@@ -52,13 +52,11 @@ class test_prepare_discover01(wttest.WiredTigerTestCase, suite_subprocess):
     scenarios = make_scenarios(types, txn_end)
 
     def test_prepare_discover01(self):
-        # Currently this test will fail because we haven't added support for
-        # packing/unpacking prepare_ts and prepared_id on checkpoint yet, so it
-        # will fail cell validation when trying to read prepared_id from disk. Re-enable this test
-        # when the feature is supported.
+        # Currently this test will crash because we try recover before setting preserve_prepared flag.
+        # Support this by moving recovery to after setting precise_checkpoint and preserve_prepared flags.
         self.conn.set_timestamp('stable_timestamp=' + self.timestamp_str(50))
         self.conn.set_timestamp('oldest_timestamp=' + self.timestamp_str(50))
-        self.skipTest('FIXME-WT-14941 Enable when packing/unpacking prepare_ts and prepared_id on checkpoint is supported')
+        # self.skipTest('FIXME-WT-15113 Enable when we support recovery from backup with preserve_prepare config')
```

Comment:
Why this test is still failing? I think it should work?

Reply:
After checkpoint with preserve_prepared, currently we will crash when trying to open a back up dir because we will try recover before setting preserve_prepared flag. However recovery depends on preserve_prepared (since we're storing prepare id now), it will crash.

We need to move read preserve_prepare and precise_checkpoint configs before trying recovery

Reply:
Thanks for the explaination